### PR TITLE
Update `get.sh` to work with apple silicon

### PR DESCRIPTION
**Pull request by TomTucka**
## Description

Update `get.sh` to perform a the same check for linux on Darwin machines
Update makefile to build a `k3sup-darwin-arm64` binary, I wasn't able to just use the `k3sup-arm64` version

## Motivation and Context

- [X] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Ran `get.sh` locally which installed `k3sup-arm64` which showed the logic in the script was working as expected but k3sup still didn't work, I was getting `zsh: exec format error: k3sup`

Added the build for `k3sup-darwin-arm64` into the makefile, installed that binary and k3sup worked as expected

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: get.sh [hunk 1]
```diff
@@ -68,7 +68,18 @@ getPackage() {
     suffix=""
     case $uname in
     "Darwin")
-    suffix="-darwin"
+        arch=$(uname -m)
```
Comment (alexellis) [inline get.sh:71]: I guess we'd want similar for [arkade](https://github.com/alexellis/arkade/releases)

File: get.sh [hunk 2]
```diff
@@ -68,7 +68,18 @@ getPackage() {
...
+        echo $arch
```
Comment (alexellis) [inline get.sh:72]: Do we need to echo the arch?

**Comment by alexellis** [comment]
This PR isn't going to work without bumping the Go version for CI since 1.15 doesn't support Darwin M1. I've made that change in b3b7106..e06bdae

Not sure, but you may need to rebase to get the effect of it.

File: get.sh (hunk 2 above)
Comment (TomTucka) [reply get.sh:72]: Removed

File: get.sh (hunk 1 above)
Comment (TomTucka) [reply get.sh:71]: PR Opened for [arkade](https://github.com/alexellis/arkade/pull/600)

**Pull request closed without merging**